staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -105,9 +105,26 @@ func NewGenericAdmissionWebhook(configFile io.Reader) (*GenericAdmissionWebhook,
 		),
 		authInfoResolver: authInfoResolver,
 		serviceResolver:  defaultServiceResolver{},
+		cache:            &clientCache{clientMap: make(map[admissionHookClientConfig]*rest.RESTClient)},
 	}, nil
 }
 
+type admissionHookClientConfig struct {
+	Service v1alpha1.ServiceReference
```

Comment:
Also add the cfg.TLSClientConfig.ServerName to the key?

Reply:
Why need seperate `ServerName `, we can get from 
`	serverName := h.ClientConfig.Service.Name + "." + h.ClientConfig.Service.Namespace + ".svc"
`